### ISSUE TITLE
hid_write REPORT_ID byte on linux/windows/mac

### ISSUE THREAD

**Issue by jonasschnelli** [signal11/hidapi#255]
I'm currently finalizing deployment of a cross platform application.
But what I can't follow is,...

Why do i need to send the REPORT_ID on mac/windows:
(pseudecode):
`hid_write(handle, 0x00 + buffer, size(buffer)+1)`

While on Linux it need to be
`hid_write(handle, buffer, size(buffer))`

Can I somehow detect if hid_write expect a REPORT_ID?

**Comment by signal11** [comment]
It should be the same. Are you using the latest from git?

**Comment by jonasschnelli** [comment]
I'm using 0.8.0-rc1.
Maybe it was fixed with https://github.com/signal11/hidapi/commit/54eb31dc16dcc67d0b689ed947bc53a038608c0e which is not in 0.8.0-rc1.

What about doing a 0.8.0 release (current master)?

**Comment by jonasschnelli** [comment]
It seems to be a difference between `hid-libusb` and `hid-hidraw` (at least on linux).

**Comment by jonasschnelli** [comment]
https://github.com/signal11/hidapi/commit/3a66d4e513ed1b1ce82b7e6fcfa30ff05598b696 this commit is un Ubuntus version but not in the 0.8.0rc1 tag. Maybe spun of a rc2?

**Comment by swt2c** [comment]
Indeed.  Ubuntu < 16.04 has a git snapshot at rev 3a66d4e.  Ubuntu 16.04+ has a git snapshot at rev d17db57.

**Comment by nvchernov** [comment]
So much thanks for this question!!! I found solution for me with your approach.
Function works great, but it behaves really unexpecting!
